core/vm/eips.go

```diff
@@ -338,13 +338,9 @@ func opExtCodeCopyEIP4762(pc *uint64, interpreter *EVMInterpreter, scope *ScopeC
 	}
 	addr := common.Address(a.Bytes20())
 	code := interpreter.evm.StateDB.GetCode(addr)
-	contract := &Contract{
-		Code: code,
-		self: AccountRef(addr),
-	}
 	paddedCodeCopy, copyOffset, nonPaddedCopyLength := getDataAndAdjustedBounds(code, uint64CodeOffset, length.Uint64())
-	if !contract.IsSystemCall {
-		statelessGas := interpreter.evm.AccessEvents.CodeChunksRangeGas(addr, copyOffset, nonPaddedCopyLength, uint64(len(contract.Code)), false)
+	if !isSystemCall(AccountRef(addr)) {
```

Comment:
You can use `scope.Contract.IsSystemCall` instead

Reply:
I got confused by the same thing at first, but this is actually incorrect: `scope.Contract.IsSystemCall` is the caller, whereas `addr` is the contract whose bytecode is copied.

Now, consider the following cases:

 * a (future) system contract has an `EXTCODECOPY` in its bytecode, and assume that it can call _any_ non-system call contract (e.g. as a factory or something). What should end up in the witness? Not the code of the system contract, as per 4762 system contracts code chunks should not be included in the witness. But the code chunks of that non-system contract that are copied should be in the witness, as it's not a system contract.
 * a regular contract copies code from a system contract, as per 4762 that code should not end up in the witness.